src/executables/compactdb_common.c

```diff
@@ -0,0 +1,271 @@
+ *    num_class_list(out): pointer to returned number of mops
+ */
+int
+get_class_mops (char **class_names, int num_class, MOP ** class_list, int *num_class_list)
+{
+  int i;
+  char downcase_class_name[SM_MAX_IDENTIFIER_LENGTH];
+  DB_OBJECT *class_ = NULL;
+
+  if (class_names == NULL || num_class <= 0 || class_list == NULL || num_class_list == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  *num_class_list = 0;
+  *class_list = (DB_OBJECT **) malloc (DB_SIZEOF (DB_OBJECT *) * (num_class));
+  if (*class_list == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  for (i = 0; i < num_class; ++i)
+    {
+      (*class_list)[i] = NULL;
+    }
+
+  for (i = 0; i < num_class; i++)
+    {
+      if (class_names[i] == NULL || strlen (class_names[i]) == 0)
+	{
+	  goto error;
+	}
+
+      sm_downcase_name (class_names[i], downcase_class_name, SM_MAX_IDENTIFIER_LENGTH);
+
+      class_ = locator_find_class (downcase_class_name);
+      if (class_ != NULL)
+	{
+	  (*class_list)[(*num_class_list)] = class_;
+	  (*num_class_list)++;
+	}
+      else
+	{
+	  printf (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_COMPACTDB, COMPACTDB_MSG_CLASS),
+		  downcase_class_name);
+
+	  printf (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_COMPACTDB, COMPACTDB_MSG_INVALID_CLASS));
+	}
+    }
+
+  return NO_ERROR;
+
+error:
+  if (*class_list)
+    {
+      free (*class_list);
+      *class_list = NULL;
+    }
+
+  if (num_class_list)
+    {
+      *num_class_list = 0;
+    }
+
+  return ER_FAILED;
+}
+
+/*
+ * get_class_mops_from_file - Get a list of class mops from specified file
+ *    return: error code
+ *    input_filename(in): input file name
+ *    class_list(out): pointer to returned list of class mops
+ *    num_class_mops(out): pointer to returned number of mops
+ */
+int
+get_class_mops_from_file (const char *input_filename, MOP ** class_list, int *num_class_mops)
+{
+  int status = NO_ERROR;
+  int i = 0;
+  FILE *input_file;
+  char buffer[DB_MAX_IDENTIFIER_LENGTH];
+  char **class_names = NULL;
+  int num_class = 0;
+  int len = 0;
+  char *ptr = NULL;
+
+  if (input_filename == NULL || class_list == NULL || num_class_mops == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  *class_list = NULL;
+  *num_class_mops = 0;
+  if (get_num_requested_class (input_filename, &num_class) != NO_ERROR)
+    {
+      return ER_FAILED;
+    }
+
+  if (num_class == 0)
+    {
+      return ER_FAILED;
+    }
+
+  input_file = fopen (input_filename, "r");
+  if (input_file == NULL)
+    {
+      perror (input_filename);
+      return ER_FAILED;
+    }
+
+  class_names = (char **) malloc (DB_SIZEOF (char *) * num_class);
+  if (class_names == NULL)
+    {
+      return ER_FAILED;
+    }
+  for (i = 0; i < num_class; i++)
+    {
+      class_names[i] = NULL;
+    }
+
+  for (i = 0; i < num_class; ++i)
+    {
+      if (fgets ((char *) buffer, DB_MAX_IDENTIFIER_LENGTH, input_file) == NULL)
+	{
+	  status = ER_FAILED;
+	  goto end;
+	}
```

Comment:
_indentation style (for all following if-statements)_:
```
if (fgets ((char *) buffer, DB_MAX_IDENTIFIER_LENGTH, input_file) == NULL)
  {
     status = ER_FAILED;
  }
```

Reply:
Indentation is changed  by executing "indent" command.
Why do you know to change indentation ?
Please let me know solution if you know.

Reply:
Sorry, I have no idea on it. Anyway, you changed indentation as I requested.